dcrlibwallet.go

```diff
@@ -2,70 +2,62 @@ package dcrlibwallet
 
 import (
 	"context"
-	"fmt"
-	"os"
 	"path/filepath"
 
-	"github.com/asdine/storm"
 	"github.com/decred/dcrwallet/errors"
 	"github.com/decred/dcrwallet/netparams"
-	"github.com/decred/dcrwallet/wallet"
-	"github.com/decred/dcrwallet/wallet/txrules"
+	wallet "github.com/decred/dcrwallet/wallet/v3"
+	"github.com/decred/dcrwallet/wallet/v3/txrules"
 	"github.com/raedahgroup/dcrlibwallet/txindex"
 	"github.com/raedahgroup/dcrlibwallet/utils"
-	"go.etcd.io/bbolt"
 )
 
-const logFileName = "dcrlibwallet.log"
+const (
+	SpendingPassphraseTypePin  int32 = 0
+	SpendingPassphraseTypePass int32 = 1
+)
+
+type WalletProperties struct {
+	WalletID               int    `storm:"id,increment"`
+	WalletName             string `storm:"unique"`
+	WalletDataDir          string
+	WalletSeed             string
+	DefaultAccount         int32
+	SpendingPassphraseType int32
+	DiscoveredAccounts     bool
+}
 
 type LibWallet struct {
-	walletDataDir string
-	activeNet     *netparams.Params
-	walletLoader  *WalletLoader
-	wallet        *wallet.Wallet
-	txDB          *txindex.DB
-	configDB      *storm.DB
-	*syncData
+	WalletProperties `storm:"inline"`
+
+	activeNet    *netparams.Params
+	walletLoader *WalletLoader
+	wallet       *wallet.Wallet
+	txDB         *txindex.DB
+
+	synced     bool
+	syncing    bool
+	waiting    bool
+	rescanning bool
 
 	shuttingDown chan bool
 	cancelFuncs  []context.CancelFunc
 }
 
-func NewLibWallet(appDataDir, walletDbDriver string, netType string) (*LibWallet, error) {
-	// initialize logger before returning any error
-	errors.Separator = ":: "
-	initLogRotator(filepath.Join(appDataDir, logFileName))
+func NewLibWallet(walletDataDir, walletDbDriver string, netType string) (*LibWallet, error) {
 
 	activeNet := utils.NetParams(netType)
 	if activeNet == nil {
-		return nil, fmt.Errorf("unsupported network type: %s", netType)
+		return nil, errors.E("unsupported network type: %s", netType)
 	}
 
 	lw := &LibWallet{
-		activeNet:     activeNet,
-		walletDataDir: filepath.Join(appDataDir, activeNet.Name),
-	}
-
-	err := os.MkdirAll(lw.walletDataDir, 0700)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create wallet db directory: %v", err)
-	}
-
-	configDbPath := filepath.Join(lw.walletDataDir, userConfigDbFilename)
-	lw.configDB, err = storm.Open(configDbPath)
-	if err != nil {
-		if err == bolt.ErrTimeout {
-			// timeout error occurs if storm fails to acquire a lock on the database file
-			return nil, fmt.Errorf("settings db is in use by another process")
-		}
-		return nil, fmt.Errorf("error opening settings db store: %s", err.Error())
+		activeNet: activeNet,
```

Comment:
`lw.WalletDataDir` is not set. Could be set here.

Reply:
`WalletDataDir` is handled by MultiWallet